### PR TITLE
MOTOR-608 Update outdated API docs

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: motor/docstrings.py [hunk 1]
```diff
@@ -201,6 +201,82 @@ async def modify_data():
...
+    option is silently ignored by the server and unique index builds
+    using the option will fail if a duplicate value is detected.
+
+  .. note:: `partialFilterExpression` requires server version **>= 3.2**
+
+  .. note:: The :attr:`~pymongo.collection.Collection.write_concern` of
+     this collection is automatically applied to this operation.
+
+  :Parameters:
+    - `keys`: a single key or a list of (key, direction)
```
Comment (ShaneHarvey) [inline motor/docstrings.py:269]: Missing session and comment parameters.

File: motor/docstrings.py [hunk 2]
```diff
@@ -1219,3 +1295,285 @@ async def coro():
         encrypted = await client_encryption.encrypt(value, ...)
         decrypted = await client_encryption.decrypt(encrypted)
 """
+
+gridfs_delete_doc = """Delete a file's metadata and data chunks from a GridFS bucket::
```
Comment (ShaneHarvey) [inline motor/docstrings.py:1299]: Let's add the session parameter to these gridfs docstrings.

**Review by ShaneHarvey — CHANGES_REQUESTED** [review]
Nice! A few small suggestions.

File: motor/docstrings.py (hunk 1 above)
Comment (blink1073) [reply motor/docstrings.py:269]: Done

File: motor/docstrings.py (hunk 2 above)
Comment (blink1073) [reply motor/docstrings.py:1299]: Done

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**